Error building mavsdk_server (LNK1149)

Got problem trying to compile mavsdk with BUILD_BACKEND = ON option. I am using VS 2017 Build Tools on Windows 10 Enterprise. I am getting the error below. Any help would be appriciated.

[build]        "D:\Work\MAVSDK\build\ALL_BUILD.vcxproj" (default target) (1) ->
[build]        "D:\Work\MAVSDK\build\src\backend\src\mavsdk_server_bin.vcxproj" (default target) (32) ->
[build]        (Link target) -> 
[build]          LINK : fatal error LNK1149: output filename matches input filename 'D:\Work\MAVSDK\build\src\backend\src\Debug\mavsdk_server.lib' [D:\Work\MAVSDK\build\src\backend\src\mavsdk_server_bin.vcxproj]
[build] 
[build]     0 Warning(s)
[build]     1 Error(s)

--- a/src/backend/src/CMakeLists.txt
+++ b/src/backend/src/CMakeLists.txt
@@ -83,9 +83,14 @@ else()
         mavsdk
     )
 
-    set_target_properties(mavsdk_server_bin PROPERTIES
-        OUTPUT_NAME mavsdk_server
-    )
+    # MSVC fails to generate the `mavsdk_server` binary while having
+    # a library called `mavsdk_server` as well. This means that with
+    # MSVC, we build `mavsdk_server_bin.exe`.
+    if(NOT MSVC)
+        set_target_properties(mavsdk_server_bin PROPERTIES
+            OUTPUT_NAME mavsdk_server
+        )
+    endif()
 
     install(TARGETS mavsdk_server_bin
         EXPORT mavsdk-targets

